File: seed/php-sdk/trace/src/Playlist/Types/Playlist.php
<?php

namespace Seed\Playlist\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Playlist\Traits\PlaylistCreateRequest;
use Seed\Core\Json\JsonProperty;

class Playlist extends JsonSerializableType
{
    use PlaylistCreateRequest;

    /**
     * @var string $playlistId
     */
    #[JsonProperty('playlist_id')]
    public string $playlistId;

    /**
     * @var string $ownerId
     */
    #[JsonProperty('owner-id')]
    public string $ownerId;

    /**
     * @param array{
     *   playlistId: string,
     *   ownerId: string,
     *   name: string,
     *   problems: array<string>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->playlistId = $values['playlistId'];
        $this->ownerId = $values['ownerId'];
        $this->name = $values['name'];
        $this->problems = $values['problems'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/playlist/types/playlist.rb
# frozen_string_literal: true

module Seed
  module Playlist
    module Types
      class Playlist < Internal::Types::Model
        field :name, -> { String }, optional: false, nullable: false

        field :problems, -> { Internal::Types::Array[String] }, optional: false, nullable: false

        field :playlist_id, -> { String }, optional: false, nullable: false

        field :owner_id, -> { String }, optional: false, nullable: false, api_name: "owner-id"
      end
    end
  end
end
